clang/include/clang/Serialization/ASTBitCodes.h

```diff
@@ -1101,7 +1101,7 @@ enum PredefinedTypeIDs {
 ///
 /// Type IDs for non-predefined types will start at
 /// NUM_PREDEF_TYPE_IDs.
-const unsigned NUM_PREDEF_TYPE_IDS = 500;
+const unsigned NUM_PREDEF_TYPE_IDS = 502;
```

Comment:
Why is 502, but 600 is bad?

Reply:
It is not the problem of 502 or 600. The test case failure in `clang/test/Modules/decl-params-determinisim.m` is because it is expecting the original TypeID before change, and updating the capacity `NUM_PREDEF_TYPE_IDS` here changes the result and so the test case needs to be updated.

Reply:
I extended `NUM_PREDEF_TYPE_IDS` to the least required, and left a note inside the test case to let future developers be aware that the test case it intertwine with the constant.

Reply:
Then your commit message should have explained this better and not said "indeterminate results"

Reply:
Yes you are right. My mistake here. Update the PR description.